rosetta/services/tx_format.go

```diff
@@ -30,7 +30,7 @@ type ContractInfo struct {
 
 // FormatTransaction for staking, cross-shard sender, and plain transactions
 func FormatTransaction(
-	tx hmytypes.PoolTransaction, receipt *hmytypes.Receipt, contractInfo *ContractInfo,
+	tx hmytypes.PoolTransaction, receipt *hmytypes.Receipt, contractInfo *ContractInfo, signed bool,
```

Comment:
why a new parameter is added here?  `signed`?  Please elaborate it in the comments.

Reply:
Because `FormatTransaction` is called by parse function, which can both parse signed and unsigned transactions. The detailed work flow is parse -> FormatTransaction -> GetNativeOperationsFromStakingTransaction -> rpcV2.NewStakingTransaction. Finally inside rpcV2.NewStakingTransaction only signed transaction can recover sender address, so I think I need this flag parameter:

```
	if signed {
		from, err := tx.SenderAddress()
		if err != nil {
			return nil, errors.New(fmt.Sprintf("get sender address error: %s", err.Error()))
		}

		fromAddr, err := internal_common.AddressToBech32(from)
		if err != nil {
			return nil, err
		}
		result.From = fromAddr
	}
```